langlib/lang.query/src/main/ballerina/helpers.bal

```diff
@@ -181,3 +215,18 @@ function consumeStream(stream<Type, CompletionType> strm) returns any|error {
         return v;
     }
 }
+
+function createImmutableTable(table<map<Type>> tbl, Type[] arr) returns table<map<Type>> & readonly = @java:Method {
+    'class: "org.ballerinalang.langlib.query.CreateImmutableType",
+    name: "createImmutableTable"
+} external;
+
+function createTableWithKeySpecifier(table<map<Type>> tbl, typedesc tableType) returns table<map<Type>> = @java:Method {
+    'class: "org.ballerinalang.langlib.query.CreateImmutableType",
+    name: "createTableWithKeySpecifier"
+} external;
+
+function createImmutableType(any mutableType) = @java:Method {
```

Comment:
Name doesn't properly reflect what we are doing here. It is the value that is being made immutable.

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/36576/commits/f28d7e8436d8a4f35acadab42d9f0fbfb09f696e.